src/main/java/com/epam/reportportal/testng/TestNGService.java

```diff
@@ -91,16 +96,23 @@ public TestNGService(Supplier<Launch> launch) {
 
 	@Override
 	public void startLaunch() {
-		this.launch.get().start();
+		Maybe<String> launchId = this.launch.get().start();
 		StepAspect.addLaunch("default", this.launch.get());
+		ITEM_TREE.setLaunchId(launchId);
 	}
 
 	@Override
 	public void finishLaunch() {
 		FinishExecutionRQ rq = new FinishExecutionRQ();
 		rq.setEndTime(Calendar.getInstance().getTime());
 		rq.setStatus(isLaunchFailed.get() ? Statuses.FAILED : Statuses.PASSED);
-		launch.get().finish(rq);
+
+		FINISH_LAUNCH_LOCK.lock();
```

Comment:
Not quite understood the logic of such lock. What for? You are not use it anywhere else. What inside `Launch.finish(rq);` should be executed serially?

Reply:
1. done
2. It doesn't matter what structure of suite is in the suite.xml file (with nested <suite-file .../>), because all nested suites reported to the same level as root suite, so "suite"->"test"->"step" is the only possible structure

Reply:
Still don't see any necessity of the lock

Reply:
removed